### ISSUE TITLE
Style/SymbolProc autocorrect breaks code

### ISSUE THREAD

**Issue by jcoyne** [rubocop-hq/rubocop#6540]
If you have code such as this:
```ruby
      builder = Nokogiri::XML::Builder.new do |xml|
        xml.events
      end
```

and you autocorrect it to:
```ruby
builder = Nokogiri::XML::Builder.new(&:events)
```

Then you get this error:
```
     ArgumentError:
       Can't create Binding from C level Proc
     # ./vendor/bundle/ruby/2.5.0/gems/nokogiri-1.8.5/lib/nokogiri/xml/builder.rb:290:in `binding'
     # ./vendor/bundle/ruby/2.5.0/gems/nokogiri-1.8.5/lib/nokogiri/xml/builder.rb:290:in `initialize'
```

Using rubocop 0.60.0

Historically this has been a very dangerous autosuggestion, see:
* #6448
* #6052 
* #5098 
* #5009
* #3071
* #2135
* #2588
* etc.

Maybe it's not something that can safely be done with static analysis.

**Comment by jcoyne** [comment]
I think they are similar, but with different failure modes. This one is specific to a C binding

**Comment by Drenmi** [comment]
> Generally the only solution I see to a problem like it is to mark the auto-correction as not safe.

Agree.
